src/GraphQL/Utilities/SchemaBuilder.cs

```diff
@@ -310,22 +314,24 @@ private void InitializeSubscriptionField(FieldConfig config, Type? parentType)
 
             if (config.ResolverAccessor != null && config.SubscriberAccessor != null)
             {
-                config.Resolver = new AccessorFieldResolver(config.ResolverAccessor, ServiceProvider);
+                config.Resolver = AutoRegisteringHelper.BuildFieldResolver(
+                    config.ResolverAccessor.MethodInfo,
+                    null, // unknown source type
+                    null, // unknown FieldType
+                    AutoRegisteringHelper.BuildInstanceExpressionForSchemaBuilder(config.ResolverAccessor.DeclaringType, ServiceProvider));
+
                 var attrs = config.ResolverAccessor.GetAttributes<GraphQLAttribute>();
                 if (attrs != null)
                 {
                     foreach (var a in attrs)
                         a.Modify(config);
                 }
 
-                if (config.SubscriberAccessor.MethodInfo.ReturnType.GetGenericTypeDefinition() == typeof(Task<>))
-                {
-                    config.AsyncSubscriber = new AsyncEventStreamResolver(config.SubscriberAccessor, ServiceProvider);
-                }
-                else
-                {
-                    config.Subscriber = new EventStreamResolver(config.SubscriberAccessor, ServiceProvider);
-                }
+                config.AsyncSubscriber = AutoRegisteringHelper.BuildEventStreamResolver(
```

Comment:
Where is Subscriber  (non async)?

Reply:
Will not exist once conversion to ValueTask is completed so I did not bother to write code for it. Besides, as it exists in this PR, synchronous subscribers will get wrapped in Task FromResult. Which is not the worst thing for a single allocation when setting up a subscription.